File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        if ( !root )
            return -1;
        std::queue<TreeNode *> treeQ;
        treeQ.push(root);
        int minLevel = 0;
        int level = 0;
        int maxSum = INT_MIN;
        while (treeQ.empty() == false) {
            int levelSize = treeQ.size();
            int levelSum = 0;
            level += 1;
            std::cout << "leve = " << level << std::endl;
            for( int i = 0; i < levelSize; i++) {
                TreeNode *tmpNode = treeQ.front();
                treeQ.pop();
                if (tmpNode->left)
                    treeQ.push(tmpNode->left);
                if (tmpNode->right)
                    treeQ.push(tmpNode->right);
                
                levelSum += tmpNode->val;
            }
            if (levelSum > maxSum) {
                maxSum = levelSum;
                minLevel = level;
            }    
        }
        
        return minLevel;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxLevelSum(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        
        if root is None:
            return 0

        maxSum = float("-inf")
        minLevel = 0
        level = 0
        
        q = collections.deque([root])
        while len(q) != 0:
            numNode = len(q)
            level += 1
            levelSum = 0
            for _ in range(numNode):
                node = q.popleft()
                levelSum += node.val
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
            if levelSum > maxSum:
                maxSum = levelSum
                minLevel = level 
            
        return minLevel
    
    
                  
            
        